README.md

```diff
@@ -109,6 +109,45 @@ The realm server uses the request accept header to determine the type of request
 | `*/*`                         | We support node-like resolution, which means that the extension is optional                                                                                                                                                                                                                                          | Used to request transpiled executable code modules                                                                                                                             |
 
+### Database
+
+The boxel system leverages a Postgres database. The Postgres database lives within a docker container, `boxel-pg`, that is started as part of `pnpm start:all`. You can manually start and stop the `boxel-pg` docker container using `pnpm start:pg` and `pnpm stop:pg`. The postgres database runs on port 5435 so that it doesn't conflict with a natively installed postgres that may be running on your system.
+
+When running tests we isolate the database between each test run by actually creating a new database for each test with a random database name (e.g. `test_db_1234567`). The test databases are dropped before the beginning of each test run.
+
+If you wish to drop the development database you can execute:
+```
+pnpm drop-db
+```
+
+You can then run `pnpm migrate up` or start the realm server to create the database again.
+
+#### DB Migrations
+When the realm server starts up it will automatically run DB migrations that live in the `packages/realm-server/migrations` folder. As part of development you may wish to run migrations manually as well as to create a new migration.
+
+To create a new migration, from `packages/realm-server`, execute:
+```
+pnpm migrate create name-of-migration
+```
+This creates a new migration file in `packages/realm-server/migrations`. You can then edit the newly created migration file with the details of your migration. We use `node-pg-migrate` to handle our migrations. You can find the API at https://salsita.github.io/node-pg-migrate. 
+
+To run the migration, execute:
+```
+pnpm migrate up
+```
+
+To revert the migration, execute:
+```
+pnpm migrate down
+```
+
+The boxel system also uses SQLite in order to run the DB in the browser as part of running browser tests (and eventually we may run the realm server in the browser to provide a local index). After you have completed your migration you then need to generate a new schema SQL file for SQLite. To generate a new SQLite schema, from `packages/realm-server`, execute:
```

Comment:
```suggestion
Boxel also uses SQLite in order to run the DB in the browser as part of running browser tests (and eventually we may run the realm server in the browser to provide a local index). We treat the Postgres database schema as the source of truth and derive the SQLite schema from it. Therefore, once you author and apply a migration, you should generate a new schema SQL file for SQLite. To generate a new SQLite schema, from `packages/realm-server`, execute:
```

Reply:
Also, would it be possible and desirable to automatically run `pnpm make-schema` after `pnpm migrate up` or `pnpm migrate down`?

Reply:
it's possible. not sure if it's desirable. one thing to keep in mind is that the application itself can run migrations (like a django app). so it's not necessarily the case that migrations will always happen as a result of running `pnpm migrate up` or `pnpm migrate down`. in order to make sure things keep in sync, though, I did add some logic when the host app starts to make sure the sqlite schema is up-to-date. My thinking was that I wanted for us to work with it for bit and based on dev feedback we can adjust as necessarily.

Reply:
I think another reason why we might want to keep the pg migration and sqlite schema generation decoupled for now is that the sqlite schema conversion is still pretty fresh. I could see scenarios where we start using a new constraint type that we haven't yet added support for in the sqlite schema generation script. in which case it might be easier for debugging if the migration and schema generation were separate.